server/image_pull.go

```diff
@@ -65,10 +68,16 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 	pullOp, pullInProcess := func() (pullOp *pullOperation, inProgress bool) {
 		s.pullOperationsLock.Lock()
 		defer s.pullOperationsLock.Unlock()
-		pullOp, inProgress = s.pullOperationsInProgress[pullArgs]
+		op, inProgress := storage.PullOperationsInProgress.Load(pullArgs)
+		if inProgress {
+			pullOp, ok = op.(*pullOperation)
+			if !ok {
+				return &pullOperation{}, inProgress
```

Comment:
I think this should be an error.  It _should_ never happen but only an error would tell us if it happens.

I really don't like `sync.Map` for having to deal with `Interface{}` (personal preference).